src/ui/views/Approval/components/SignTx.tsx

```diff
@@ -232,6 +232,182 @@ export const TxTypeComponent = ({
   return <></>;
 };
 
+const getRecommendGas = async ({
+  gas,
+  wallet,
+  tx,
+  chainId,
+}: {
+  gas: number;
+  wallet: ReturnType<typeof useWallet>;
+  tx: Tx;
+  chainId: number;
+}) => {
+  if (gas > 0) {
+    return gas;
+  }
+  if (Number(tx.gasLimit || tx.gas) > 0) {
+    return Number(tx.gasLimit || tx.gas);
+  }
+  const res = await wallet.openapi.historyGasUsed({
+    tx: {
+      ...tx,
+      nonce: tx.nonce || '0x1', // set a mock nonce for explain if dapp not set it
+      data: tx.data,
+      value: tx.value || '0x0',
+      gas: tx.gas || '', // set gas limit if dapp not set
+    },
+    user_addr: tx.from,
+  });
+  if (res.gas_used > 0) {
+    return res.gas_used;
+  }
+  const chain = Object.values(CHAINS).find((item) => item.id === chainId);
+  if (!chain) {
+    throw new Error('chain not found');
+  }
+  const block = await wallet.requestETHRpc(
+    {
+      method: 'eth_getBlockByNumber',
+      params: ['latest', true],
```

Comment:
```suggestion
      params: ['latest', false],
```
Why we need details for each transaction?